### PR TITLE
Add a "flatten" transform example for JSON data.

### PR DESCRIPTION
**Pull request by voutilad**
Based on the ["Flatten" Apache Kafka SMT](https://docs.confluent.io/platform/current/connect/transforms/flatten.html) for flattening nested JSON structures.

Requires building with "asyncify" scheduler, currently, because of the streaming JSON parser. (Looking to replace it with a streaming parser that doesn't use Go routines...but want to get this imported first.)

### CODE REVIEW

File: data-transforms/flatten/transform.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	}
+	fmt.Println("Using delimitter: ", delim)
+
+	buffer := bytes.NewBuffer(make([]byte, 1024))
+	fn := func(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+		return doFlatten(e, buffer, delim)
+	}
+
+	redpanda.OnRecordWritten(fn)
+	fmt.Println("Registered flatten function")
```
Comment (rockwotj) [inline data-transforms/flatten/transform.go:28]: In the latest version of the broker, this should never be printed. The above call is a blocking one.

File: data-transforms/flatten/transform.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+name: flatten
+input-topic: ""
+output-topic: ""
+language: tinygo
```
Comment (rockwotj) [inline data-transforms/flatten/transform.yaml:4]: if you use `tinygo-with-goroutines` then we'll automatically add the asyncify bit: 

https://github.com/redpanda-data/redpanda/pull/15103

File: data-transforms/flatten/transform.go (hunk 1 above)
Comment (voutilad) [reply data-transforms/flatten/transform.go:28]: Do we have a non-blocking log api exposed?

Comment (rockwotj) [reply data-transforms/flatten/transform.go:28]: what are you asking for? The problem is that OnRecordWritten blocks forever, logging should work fine

Comment (voutilad) [reply data-transforms/flatten/transform.go:28]: Oh that wasn't clear. I thought your comment was saying `fmt.Println` is a blocking call.

File: data-transforms/flatten/transform.yaml (hunk 2 above)
Comment (voutilad) [reply data-transforms/flatten/transform.yaml]: Thanks. Updated that.

**Comment by voutilad** [comment]
Realized my README.md wasn't imported. (Was juggling some different branches...) Updated my branch and accounted for @rockwotj's feedback.

File: data-transforms/flatten/transform.go (hunk 1 above)
Comment (rockwotj) [reply data-transforms/flatten/transform.go]: Ah sorry 'bout that!

**Review by jrkinley — APPROVED** [review]

**Pull request merged**